autoscale/src/main/java/com/sequenceiq/periscope/monitor/executor/EvaluatorExecutorRegistry.java

```diff
@@ -25,88 +25,79 @@
     @Inject
     private Clock clock;
 
-    private final Map<Key, Element> submittedEvaluators = new ConcurrentHashMap<>();
+    private final Map<Element, Element> submittedEvaluators = new ConcurrentHashMap<>();
```

Comment:
The key and value are the same.

Reply:
I tried to replace, but makes code complicated, because other collection implementations can't add new or return the existing value, so the solution requires many workarounds to make it thread safe.